tests/php/Core/EnvironmentTest.php

```diff
@@ -66,4 +67,87 @@ public function testGetVariables()
         $this->assertEquals('fail', $vars['test']);
         $this->assertEquals('global', $GLOBALS['test']);
     }
+
+    public function provideHasEnv()
+    {
+        $setAsOptions = [
+            '.env',
+            '_ENV',
+            '_SERVER',
+            'putenv',
+        ];
+        $valueOptions = [
+            true,
+            false,
+            null,
+            0,
+            1,
+            1.75,
+            '',
+            '0',
+            'some-value',
+        ];
+        $scenarios = [];
+        foreach ($setAsOptions as $setAs) {
+            foreach ($valueOptions as $value) {
+                $scenarios[] = [
+                    'setAs' => $setAs,
+                    'value' => $value,
+                    'expected' => true,
+                ];
+            }
+        }
+        $scenarios[] = [
+            'setAs' => null,
+            'value' => null,
+            'expected' => false,
+        ];
+        return $scenarios;
+    }
+
+    /**
+     * @dataProvider provideHasEnv
+     */
+    public function testHasEnv(?string $setAs, $value, bool $expected)
+    {
+        $name = '_ENVTEST_HAS_ENV';
+
+        // Set the value
+        switch ($setAs) {
+            case '.env':
+                Environment::setEnv($name, $value);
+                break;
+            case '_ENV':
+                $_ENV[$name] = $value;
+                break;
+            case '_SERVER':
+                $_SERVER[$name] = $value;
+                break;
+            case 'putenv':
+                if ($value === null) {
+                    $val = 'null';
+                } elseif (is_bool($value)) {
+                    $val = $value ? 'true' : 'false';
+                } else {
+                    $val = (string)$value;
+                }
```

Comment:
This can be simplified to..., but I appreciate it might not be as clear as to the mechanics of the conversion.
```suggestion
                $val = is_string($value) ? $value : json_encode($value);
```

Reply:
Ooooh, nice! Done.